server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1993,4 +1992,23 @@ ON CONFLICT (real_id, \"date\", year)
             "2030| 99998.0"
         );
     }
+
+    @Test
+    public void test_non_deterministic_sub_column_unused_in_targets() {
```

Comment:
Is it possible to have a unit test instead?

Reply:
Actually in the current state would be enough (to test that `addGeneratedValues` doesn't throw an exception) and primary stored properly should be addressed by the first commit. 

But after seeing https://github.com/crate/crate/pull/15272#discussion_r1438084227  I think it's  a valid point and then I need an integration test.

Reply:
If possible it would be nice to have a unit test on top though.

Reply:
Added a unit test - confirmed that it fails if I revert changes in the `addGeneratedValues` back to master state.